geonode/qgis_server/helpers.py

```diff
@@ -468,6 +469,98 @@ def legend_url(layer, layertitle=False, style=None, internal=True):
     return url
 
 
+def qgs_url(layer, style=None, internal=True):
+    """Construct QGIS Server url to fetch QGS.
+
+    :param layer: Layer to use
+    :type layer: Layer
+
+    :param style: Layer style to choose
+    :type style: str
+
+    :param internal: Flag to switch between public url and internal url.
+        Public url will be served by Django Geonode (proxified).
+    :type internal: bool
+
+    :return: QGIS Server request url for QGS
+    :rtype: str
+    """
+    qgis_server_url = qgis_server_endpoint(internal)
+
+    ogc_url = reverse('qgis_server:layer-request',
+                      kwargs={'layername': layer.name})
+    url = settings.SITEURL + ogc_url.replace("/", "", 1)
```

Comment:
Use urljoin so you don't have to use `.replace("/", "", 1)`.
I didn't quite understand, what is this url for? Can it be replaced using `qgis_server_endpoint` ?